src/dsp_tools/commands/xmlupload/models/ontology_lookup_models.py

```diff
@@ -75,3 +77,78 @@ def _get_all_properties_from_json(onto_json: list[dict[str, Any]]) -> list[str]:
 
 def _remove_prefixes(ontology_elements: list[str]) -> list[str]:
     return [x.split(":")[1] for x in ontology_elements]
+
+
+AllowedEncodings = Literal["utf8", "xml"]
+
+
+@dataclass
+class TextValueData:
+    resource_id: str
+    property_name: str
+    encoding: AllowedEncodings
+
+
+@dataclass
+class PropertyTextValueEncodingTypes:
+    """
+    This class contains the information
+    which properties have which type of encoding for a TextValue property in the ontology.
+    """
+
+    formatted_text: set[str] = field(default_factory=set)
+    unformatted_text: set[str] = field(default_factory=set)
+
+
+def get_text_value_properties_and_formatting_from_json(
+    onto_json_dict: dict[str, list[dict[str, Any]]], default_onto: str
+) -> PropertyTextValueEncodingTypes:
+    """
+    This function takes a dict with the names and ontology json of the project.
+    It filters out the property with TextValues as object
+    and separates them into properties with formatted and unformatted text
+
+    Args:
+        onto_json_dict: dict with the project ontologies
+        default_onto: name of the default ontology
+
+    Returns:
+        Look-up containing the properties separated according to the formatting
+    """
+    all_props = []
+    for onto_json in onto_json_dict.values():
```

Comment:
`onto_json` is a list of dicts. For me this doesn't make sense. A list can never be an ontology. So what does this list contain?

Reply:
This is the `@graph` section of the ontology serialised into json-ld (we get actually standard, valid json-ld from the api)

```
{
    "rdfs:label": "The knora-api ontology in the complex schema",
    "@graph": [
        { property } ,
        { class } ,
    ]
}
```

This is a list of dicts. Granted removing all the context that comes after the list makes it an invalid json-ld (no declared prefixes). 

```
    "@context": {
        "rdf": "http://www.w3.org/1999/02/22-rdf-syntax-ns#",
        "knora-api": "http://api.knora.org/ontology/knora-api/v2#",
        "owl": "http://www.w3.org/2002/07/owl#",
        "salsah-gui": "http://api.knora.org/ontology/salsah-gui/v2#",
        "rdfs": "http://www.w3.org/2000/01/rdf-schema#",
        "xsd": "http://www.w3.org/2001/XMLSchema#"
    }
```

This we do because they are not actually relevant to us, because we do not treat it like rdf.
But it very much is the ontology in a rdf graph.